packages/block-editor/src/components/block-mover/index.js

```diff
@@ -44,24 +44,60 @@ export class BlockMover extends Component {
 	}
 
 	render() {
-		const { onMoveUp, onMoveDown, isFirst, isLast, isDraggable, onDragStart, onDragEnd, clientIds, blockElementId, blockType, firstIndex, isLocked, instanceId, isHidden, rootClientId } = this.props;
+		const { onMoveUp, onMoveDown, orientation, isRTL, isFirst, isLast, isDraggable, onDragStart, onDragEnd, clientIds, blockElementId, blockType, firstIndex, isLocked, instanceId, isHidden, rootClientId } = this.props;
 		const { isFocused } = this.state;
 		const blocksCount = castArray( clientIds ).length;
 		if ( isLocked || ( isFirst && isLast && ! rootClientId ) ) {
 			return null;
 		}
 
+		const getArrowIcon = ( hdir ) => {
+			let arrow;
+			switch ( hdir ) {
+				case 'left':
+					arrow = isRTL ? rightArrow : leftArrow;
+					arrow = orientation === 'horizontal' ? arrow : upArrow;
+					break;
+
+				case 'right':
+					arrow = isRTL ? leftArrow : rightArrow;
+					arrow = orientation === 'horizontal' ? arrow : downArrow;
+					break;
+
+				default:
+					break;
+			}
+			return arrow;
+		};
```

Comment:
nit: Maybe it's just me, but I find the `switch` unnecessarily confusing. Would a regular `if` be clearer?

```js
const getArrowIcon = ( moveDirection ) => {
	if ( moveDirection === 'up' ) {
		if ( orientation === 'horizontal' ) {
			return isRTL ? rightArrow : leftArrow;
		} else {
			return upArrow;
		}
	} else if ( moveDirection === 'down' ) {
		if ( orientation === 'horizontal' ) {
			return isRTL ? leftArrow : rightArrow;
		} else {
			return downArrow;
		}
	}
	return null;
};
```

Reply:
mind if I copy paste this? no? I knew it!